pymc3/distributions/timeseries.py

```diff
@@ -464,6 +464,69 @@ def logp(self, x):
     def _distr_parameters_for_repr(self):
         return ["mu", "cov"]
 
+    def random(self, point=None, size=None):
+        """
+        Draw random values from MvGaussianRandomWalk.
+
+        Parameters
+        ----------
+        point: dict, optional
+            Dict of variable values on which random values are to be
+            conditioned (uses default point if not specified).
+        size: int or tuple of ints, optional
+            Desired size of random sample (returns one sample if not
+            specified).
+
+        Returns
+        -------
+        array
+
+
+        Examples
+        -------
+        Create one sample from a 2-dimensional Gaussian random walk with 10 timesteps::
+
+            mu = np.array([1.0, 0.0])
+            cov = np.array([[1.0, 0.0], [0.0, 2.0]])
+            sample = MvGaussianRandomWalk(mu, cov, shape=(10, 2)).random()
+
+        Create three samples from a 2-dimensional Gaussian random walk with 10 timesteps::
+
+            mu = np.array([1.0, 0.0])
+            cov = np.array([[1.0, 0.0], [0.0, 2.0]])
+            sample = MvGaussianRandomWalk(mu, cov, shape=(10, 2)).random(size=3)
+
+        Create four samples from a 2-dimensional Gaussian random walk with 10
+        timesteps, indexed with a (2, 2) array::
+
+            mu = np.array([1.0, 0.0])
+            cov = np.array([[1.0, 0.0], [0.0, 2.0]])
+            sample = MvGaussianRandomWalk(mu, cov, shape=(10, 2)).random(size=(2, 2))
+        """
+
+        time_steps = self.shape[0]
+        size = to_tuple(size)
+
+        # for each draw specified by the size input, we need to draw time_steps many
+        # samples from MvNormal.
+        size_time_steps = size + to_tuple(time_steps)
+
+        multivariate_samples = self.innov.random(point=point, size=size_time_steps)
```

Comment:
So, you are passing spatial dims during initialization of MvNormal and pass the `time_dims` along with `size` here while drawing samples. 

Let's say I want to model just a single time series and passed in the shape just (spatial_dims) instead of (1, spatial_dims) to `MvGaussianRandomWalk`. Then with the shape handing above, multivariate samples will be drawn with shape (sample_shape, spatial_dims, spatial_dims).

I would say better it would be to pass 
1. (time_dims, spatial_dims) during initialization of MvNormal.
2. only (size) while drawing samples to take care of corner cases for us.

Reply:
Great point. I've made some more changes that I will push soon. I'm finding an error being thrown from `MvGaussianRandomWalk.logp` when I use `shape = 2` instead of `shape = (1, 2)`. Is passing just an int for shape supported by `MvGaussianRandomWalk`?

Reply:
Looking at the logp code, I think it only supports two dimensional shape as input.
How about adding something similar as MvNormal logp? as the latter handles single dimensional shape as well.

https://github.com/pymc-devs/pymc3/blob/3cfee77fda041483e86d9c61fa1d1ada9380cdb3/pymc3/distributions/multivariate.py#L114-L116

Reply:
> None for that matter

This is an interesting case. MvNormal expects any of cov, chol or tau to be a square matrix. So, I expect to specify at least one dimension when initialising MvNormal. Same reasoning applies to MvGaussianRandomWalk as well.

Reply:
Okay, I think that worked! Thanks for the help, @Sayam753. The current functionality when multiple time steps are requested is to center the walk so that it starts at 0. Should this also be done when no time dimension is passed in? It would seem silly to write it that way, since then passing something like `shape = 2` would always give you an array of zeros. My guess is that if shape has only one dimension, then the user wants a sample from a multivariate normal, so that's what should actually get returned.

Reply:
Good point. Let's not center the multivariate samples at zero in that case. If I understood correctly, the cumulative sum operation is performed irrespective whether there is a single dimension or twice?

Reply:
Well, the way I rewrote it (which I haven't pushed yet) only computes a cumulative sum if there is a time axis given as well. Otherwise it skips that step too and just returns the normal samples.